walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,94 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> eventsSortedByEnd = sortEvents(events, false);
+
+    clearIrrelevantEvents(eventsSortedByEnd, request.getAttendees());
+
+    return findAvailableTimeSlots(eventsSortedByEnd, request.getDuration());
+  }
+
+  private Collection<TimeRange> findAvailableTimeSlots(
+      Collection<Event> eventsSortedByEnd, long minDuration) {
+    Collection<TimeRange> options = new ArrayList<>();
+
+    if (eventsSortedByEnd.isEmpty()) {
+      options.add(TimeRange.WHOLE_DAY);
+    } else {
```

Comment:
You can take away the else {...} brackets and instead add a `return options` at the end of the if statement.

Reply:
Will do this as well.